src/lib/Translation/Extractor/JavaScriptFileVisitor.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Translation\Extractor;
+
+use JMS\TranslationBundle\Logger\LoggerAwareInterface;
+use JMS\TranslationBundle\Model\FileSource;
+use JMS\TranslationBundle\Model\Message;
+use JMS\TranslationBundle\Model\MessageCatalogue;
+use JMS\TranslationBundle\Translation\Extractor\FileVisitorInterface;
+use Peast\Peast;
+use Peast\Syntax\Exception;
+use Peast\Syntax\Node;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use SplFileInfo;
+
+class JavaScriptFileVisitor implements FileVisitorInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    // Example match: /** @Desc("Default text") */
+    const DESC_PATTERN = '/\*?\s*@Desc\(["\'](.*)["\']\)\s*/';
```

Comment:
I prefer full match instead of partial one. Right now it will match anything that contains `@Desc("Something")`, even if it is part of something like: `/* @todo: It would be good to add something like @Desc("Translated text") here when we get proper translation */`. It should look for exact match of a comment only with `@Desc` element. (Proper pattern is on the way)